scheduling/shortest_deadline_first.py

```diff
@@ -0,0 +1,70 @@
+"""
+Earliest Deadline First (EDF) Scheduling Algorithm
+
+This code implements the Earliest Deadline First (EDF)
+scheduling algorithm, which schedules processes based on their deadlines.
+If a process cannot meet its deadline, it is marked as "Idle."
+
+Reference:
+https://www.geeksforgeeks.org/
+earliest-deadline-first-edf-cpu-scheduling-algorithm/
+
+Author: Arunkumar
+Date: 14th October 2023
+"""
+
+
+def earliest_deadline_first_scheduling(
+    processes: list[tuple[str, int, int, int]]
```

Comment:
Instead of implementing the processes as regular tuples, could you modify your code to use [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple)? This would allow for stricter enforcement of valid input while also making the code more readable.